geode-wan/src/main/java/org/apache/geode/cache/client/internal/locator/wan/LocatorHelper.java

```diff
@@ -46,13 +46,42 @@ public static boolean addLocator(int distributedSystemId, DistributionLocatorId
     Set<DistributionLocatorId> existingValue =
         allLocatorsInfo.putIfAbsent(distributedSystemId, locatorsSet);
     if (existingValue != null) {
-      if (!existingValue.contains(locator)) {
+      if (locator.getMemberName() != null) {
```

Comment:
Mario, I think you still need need to check for an empty string and treat that the same as a null.  Maybe it would be easier to have the default value be an empty string in DistributionLocatorId.  Then you could check for an empty string and not need a null check.

Reply:
Hi, I added this part to cover empty string

  public String getMemberName() {
    if (this.membername != null && this.membername == "")
      return null;

    return this.membername;
  }

, but can also add to have empty string instead of null.